assets/js/modules/search-console/util/report-date-range-args.test.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * Tests for reporting date range utilities.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { generateDateRangeArgs } from './report-date-range-args';
+
+describe( 'Search Console reporting date range arguments', () => {
+	describe( 'generateDateRangeArgs', () => {
+		const allArgs = {
+			startDate: '2020-12-18',
+			endDate: '2021-01-14',
+		};
+
+		describe( 'if `startDate` or `endDate` arguments are not provided', () => {
+			it( 'should return args untouched', () => {
```

Comment:
This is nowhere in the ACs/IBs and seems unexpected to me. Let's have it return `undefined` if the args aren't present. Or use `invariant` to warn against improper usage—either works for me.

Since it's a new function that isn't used anywhere else it might be nicer to use `invariant` when the args aren't a valid format. 🤔

Reply:
Bit of a misunderstanding I think: I picked this up from https://github.com/google/site-kit-wp/commit/58101f479d9cec82fe81b86a4234edc3161e3a6d , and understood / thought that the function had been written to spec. When I came to write the tests I included this behaviour. I did think it was unusual, but I'll just ask next time 🙂